crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -153,9 +153,12 @@ impl<'interner> TypeChecker<'interner> {
                 let start_span = self.interner.expr_span(&for_expr.start_range);
                 let end_span = self.interner.expr_span(&for_expr.end_range);
 
-                let mut unify_loop_range = |actual_type, span| {
+                let mut unify_loop_range = |actual_type: &Type, span: Span| {
```

Comment:
See https://github.com/noir-lang/noir/pull/1583

Reply:
1583 is in the merge queue, I've tentatively merged it into this branch